src/cloudai/configurator/env_params.py

```
@@ -214,7 +270,16 @@ def from_test(cls, test: "TestDefinition") -> Optional["EnvParams"]:
             value = getattr(test.cmd_args, name, None)
             if not isinstance(value, list):
                 continue
-            params[name] = EnvParam(candidates=value, weights=spec.weights, encoding=spec.encoding)
+            encoding = spec.encoding
+            if encoding is None:
+                encoding = _infer_encoding(value)
+            elif isinstance(encoding, LogEncoding):
+                for c in value:
+                    if not isinstance(c, (int, float)) or isinstance(c, bool):
+                        raise ValueError(f"LogEncoding for '{name}' requires numeric candidates, got {type(c).__name__}")
```

Ruff (0.15.21)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)